packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -771,6 +844,36 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
     return _currentBottomSheet;
   }
 
+  // FAB API
+  AnimationController _fabMoveController;
+  FabPosition _previousFabPosition;
+  FabPosition _fabPosition;
+
+  void _moveFab(final FabPosition newPosition) {
+    void updatePosition() {
+      setState(() {
+        _previousFabPosition = _fabPosition;
+        _fabPosition = newPosition;
+        print(
+          'Moving fab from $_previousFabPosition to $_fabPosition'
+        );
+      });
+      _fabMoveController.forward(from: 0.0);
+    }
+
+    // If the fab is moved while still animating, schedule the new transition for the end of the current animation.
+    if (_fabMoveController.isAnimating) {
+      new Future<Null>.delayed(_fabMoveController.duration - _fabMoveController.lastElapsedDuration).then((_) {
+        updatePosition();
+      });
+    } else {
+      updatePosition();
+    }
```

Comment:
i don't understand this

Reply:
I've reworked this section to schedule animations more intelligently.